.github/workflows/my-ci-pipeline.yml

```diff
@@ -27,4 +28,44 @@ jobs:
         run: docker build --target test --tag to-do-test-image .
 
       - name: test
-        run: docker run --env-file .env.test to-do-test-image
\ No newline at end of file
+        run: docker run --env-file .env.test to-do-test-image
+
+  build:
+    needs: test
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main'}}
+    name: Create Docker image and push to docker hub
+    runs-on: ubuntu-latest
+    env:
+      IMAGE_TAG: ${{ github.sha }}
+    
+    steps:  
+      - uses: actions/checkout@v2        
+     
+      - run: echo "Publishing"
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Build Docker image
+        run: docker build --target production --tag philw10/todo-app:$IMAGE_TAG .
+
+      - name: Push images to Docker hub
+        run: docker push philw10/todo-app:$IMAGE_TAG    
```

Comment:
Nice work getting this working! One nice addition would potentially be to make it push _both_ the image tag and the default (`:latest`) tag so that we have one version of the image that automatically updates, and others that can be independently versioned

This can be done by just tagging the same build twice, and then pushing both independently (the second push should recognise the shared layers and be pretty quick)

(Hint: this will prove helpful later when Azure wants to pull the latest version on every deploy!)